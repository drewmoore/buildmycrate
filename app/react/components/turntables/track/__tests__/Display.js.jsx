/* globals describe test expect */
import React                   from 'react';
import renderer                from 'react-test-renderer';
import TurntablesTrackDisplay  from '../Display.js.jsx';
import Fixtures                from '../../../../__tests__/helpers/fixtures.js.es6';
import '../../../../__tests__/helpers/initialize.js.es6';

describe('TurntablesTrackDisplay', () => {
  const track = Fixtures.tracks.fortySixAndTwo;
  const props = {
    imageDisplay: track.artworkUrl,
    bpmDisplay:   'BPM: 120',
    keyDisplay:   'Key: dmin'
  };
  Object.assign(props, track);
  const template = (<TurntablesTrackDisplay {...props} />);

  describe('DOM Snapshot', () => {
    const tree = renderer.create(template).toJSON();

    test('should match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
