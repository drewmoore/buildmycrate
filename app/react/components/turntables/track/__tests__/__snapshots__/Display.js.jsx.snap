// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurntablesTrackDisplay DOM Snapshot should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-3"
  >
    <img
      alt="Track"
      className="turntable-track-image"
      src={undefined}
    />
  </div>
  <div
    className="col-xs-9"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-9 fat-bottom"
      >
        <span>
          Forty-Six and Two
        </span>
      </div>
      <div
        className="col-xs-3 fat-bottom text-right"
      >
        <span>
          BPM: 120
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-9"
      >
        <span />
      </div>
      <div
        className="col-xs-3 text-right"
      >
        <span>
          Key: dmin
        </span>
      </div>
    </div>
  </div>
</div>
`;
